aws/request/http_request_test.go

```diff
@@ -0,0 +1,49 @@
+// +build go1.5
+
+package request_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/awstesting"
+)
+
+func TestRequestCancel(t *testing.T) {
+	c := make(chan struct{})
+
+	s := awstesting.NewMockClient()
+	s.Handlers.Validate.Clear()
+	out := &testData{}
+	r := s.NewRequest(&request.Operation{Name: "Operation"}, nil, out)
+	r.HTTPRequest.Cancel = c
+	close(c)
```

Comment:
would be helpful to have an additional test that closes the `HTTPRequest.Cancel` channel in a goroutine after Send is called.